File: java/src/Behavioral/Mediator/DvdLowercaseTitle.java
package Behavioral.Mediator;


//DvdLowercaseTitle.java - One of Two Concrete Colleagues or Mediatees

public class DvdLowercaseTitle extends DvdTitle {
   private String LowercaseTitle;
   private DvdMediator dvdMediator;
    
   public DvdLowercaseTitle(String title, DvdMediator dvdMediator) {
       this.setTitle(title);
       resetTitle();
       this.dvdMediator = dvdMediator;
       this.dvdMediator.setLowercase(this);
   }    
   
   public DvdLowercaseTitle(DvdTitle dvdTitle, 
                            DvdMediator dvdMediator) {
       this(dvdTitle.getTitle(), dvdMediator);
   }     
   
   public void resetTitle() {
       this.setLowercaseTitle(this.getTitle().toLowerCase());
   }
   public void resetTitle(String title) {
       this.setTitle(title);
       this.resetTitle();
   }
   
   public void setSuperTitleLowercase() {
       this.setTitle(this.getLowercaseTitle());
       dvdMediator.changeTitle(this);
   }
   
   public String getLowercaseTitle() {
       return LowercaseTitle;
   }
   private void setLowercaseTitle(String LowercaseTitle) {
       this.LowercaseTitle = LowercaseTitle;
   }
}

File: python/src/Behavioral/Mediator/DvdLowercaseTitle.py


#//DvdLowercaseTitle- One of Two Concrete Colleagues or Mediatees

define ['Behavioral/Mediator/DvdTitle'
],(
DvdTitle
) ->

	class DvdLowercaseTitle extends DvdTitle
		LowercaseTitle : null
		dvdMediator : null

		constructor : (title, dvdMediator) ->
			@setTitle( if title instanceof DvdTitle then title.getTitle() else title)
			@resetTitle()
			@dvdMediator = dvdMediator
			@dvdMediator.setLowercase(@)

		resetTitle : (title) ->
			if title
				@setTitle(title)
			@setLowercaseTitle(@getTitle().toLowerCase())

		setSuperTitleLowercase : ->
			@setTitle(@getLowercaseTitle())
			@dvdMediator.changeTitle(@)

		getLowercaseTitle : ->
			@LowercaseTitle

		setLowercaseTitle : (aLowercaseTitle)->
			@LowercaseTitle = aLowercaseTitle
